src/mrinufft/operators/base.py

```diff
@@ -115,136 +117,284 @@ class or instance of class if args or kwargs are given.
         else:
             # instance will be created later
             operator = partial(operator.with_autograd, wrt_data, wrt_traj)
+
     return operator
 
 
 def with_numpy(fun):
     """Ensure the function works internally with numpy array."""
+    if _ismethod(fun):
+        data_arg_idx = 1
+    else:
+        data_arg_idx = 0
```

Comment:
```suggestion
    data_arg_idx = _ismethod(fun)
```
boolean, integers, its the same for python

Reply:
Here I want to select the index of the old "data" argument assuming it is the first (for standalone functions and staticmethods) or second (class or instance methods) positional argument. Using booleans, `args[False]` would return `[]`, while `args[True]` is the same as `args[None,...]` (apparently). If we want to avoid the `if - else` statement, we should do something like `data_arg_idx = int(not(_ismethod(fun))`, but I am not sure it is that readable...

Reply:
Ah good catch ! I am okay with `data_arg_idx = int(not(_ismethod(fun))` :)

Reply:
My bad, args is obviously a list and not a numpy array, so indexing with booleans would actually work as intended (also, I have no idea why I thought I needed a "not", sorry)